tpi_rs/src/app/bmc_application.rs

```diff
@@ -53,12 +56,16 @@ impl BmcApplication {
             .register_key(USB_CONFIG, &UsbConfig::UsbA(NodeId::Node1, false))
             .build()
             .await?;
+        let serial = SerialConnections::new()?;
+        // This Arc<Mutex<>> is used only to retain interior mutability
+        let serial = Arc::new(Mutex::new(serial));
```

Comment:
the mutability handling can be tidied up a bit.

You implemented a near interior mutability inside `Handler`, only the `sink` member is not wrapped. Since all of your members are wrapped with mutexes, you don't gain any advantage of implementing interior mutability. In this case, I would move the handling one layer up to the `SerialConnections` object. It will look like this:

```rust 
#[derive(Debug)]
pub struct SerialConnections {
    handlers: Vec<Mutex<Handler>>,
}
```
This enables you to remove the `Arc<Mutex<>>` inside of the `Handler` and to remove
the `Arc<Mutex<SerialConnections>>` inside `BmcApplication`.

Note: if you tend to read a lot and only write sporadically to an Object wrapped in Mutex, it is more performant to use a `RwLock`